lib/cipherstash/client/auth0_device_code_credentials.rb

```diff
@@ -148,7 +148,12 @@ def get_device_code_polling_info
       def prompt_user(polling_info)
         code = polling_info[:user_code]
 
-        Launchy.open polling_info[:verification_uri_complete]
+        begin
+          Launchy.open polling_info[:verification_uri_complete]
+        rescue Launchy::Error
+          # We could be running in a headless environment, in which case, the user should just open the url in a browser on a separate machine
+          @logger.debug("Auth0DeviceCodeCredentials#prompt_user") { "Unable to launch url in browser" }
+        end
```

Comment:
Will they know which url to open/go to?

It might be worth adding some instructions in the print out, on what to do to give more context.

Reply:
Can you use the polling info to print out the url `polling_info[:verification_uri_complete]`?

Reply:
Apparently, we have that line already [here](https://github.com/cipherstash/ruby-client/pull/21/files#diff-0a69135af67fae41ce4f6996886f645bf24d81309bf255f3d91ec006453e9f4eR159)

This is what that looks like.
![image](https://user-images.githubusercontent.com/18181377/186579864-798aca05-36fa-486a-b55a-d84cdfa3222c.png)

So, seems like the line has always been there? We could change the copy if needed.

Reply:
Looking at the screenshot in the ticket, it doesn't get to that part. The task aborts before that step. So it doesn't look like that part displays?

Reply:
offline convo, i didn't realise that the above is what displays now because we catch the error.